### PR TITLE
testing: add snapshot and trybuild tests

### LINKED ISSUE #692: Add CI check to ensure our markdown snippets are up-to-date
- We use markdown snippets in various parts of our docs 
- Doc snippets are things like `{{#include ../../README.md:95:321}}`
- They help up not have to duplicate/copy-paste docs everywhere
- However, when we make docs changes, it's easy to forget to update the docs snippets
- We should add a CI check to ensure that docs snippets are always up-to-date

Additional Context
- Maybe we hash a snippet and compare snippet hashes on CI?

### LINKED ISSUE #737: Auto-update `forc-index` documentation
Related to [this](https://github.com/FuelLabs/fuel-indexer/pull/732#discussion_r1158772442) comment, we should be able to auto update our docs based on the output of `forc-index build --help` and all other forc-index subcommands.

### LINKED ISSUE #1036: Add `insta` test coverage for `fuel-indexer-schema`
- We use [`insta`](https://docs.rs/insta/latest/insta/) for snapshot testing
- We have a bit of coverage in `fuel_indexer_schema::lib` but we're not covering all types/cases
- Since these modules (specifically the `FtColumn` type) are so important, we really should have snapshot testing for all of this
- Tests should cover any `panic!` cases as well

### LINKED ISSUE #1229: Remove `local_repository_root` hack
- [It is here](https://github.com/FuelLabs/fuel-indexer/blob/develop/packages/fuel-indexer-lib/src/utils.rs#L42)
- It is apparently required for the `trybuild` tests to pass
- We should adjust the `trybuild` tests if necessary so that this hack is no longer required

### PR DESCRIPTION
**Pull request by ra0x3**
Thanks for opening a PR with the Fuel Indexer project. Please review the **Checklist** below and ensure you've completed all of the necessary steps to make this PR review as painless as possible.

### Checklist
- [ ] Ensure your top-level commit message is in line with our [contributor guidelines](./CONTRIBUTING.md).
- [ ] Please add proper labels.
- [ ] If there is an issue associated with this PR, please link the issue (right-hand sidebar)
- [ ] If there is not an issue associated with this PR, add this PR to the "Fuel Indexer" project (right-hand sidebar)
- [ ] Please allow Codeowners at least 24 hours to do a first-pass review.
- [ ] Please add thoroughly detailed testing steps below.
- [ ] Please keep your Changelog message short and sweet.

### Description

- PR tackles a few issues related to testing
  - Adds more trybuild tests for different indexer states (mainly errors)
  - Adds more insta/snapshot tests (for our commands, and for indexer schema)
  - Cleans up a lot of clutter in the tests (removing unnecessary asset files and such)
  - Removes the plugin tests from CI

### Testing steps

- Lots of tests added so CI should pass
- No indexer functionality was changed, so nothing to manually test

### Changelog

- testing: add snapshot and trybuild tests

### CODE REVIEW

File: packages/fuel-indexer-lib/src/defaults.rs [hunk 1]
```diff
@@ -82,6 +82,15 @@ pub const AUTH_ENABLED: bool = false;
 /// Amount of time (seconds) before expiring token (if JWT scheme is specified).
 pub const JWT_EXPIRY_SECS: usize = 2592000;
 
+/// JWT secret (should not be used in production)
+pub const JWT_SECRET: &str = "abcdefghijklmnopqrstuvwxyz1234567890";
```
Comment (lostman) [inline packages/fuel-indexer-lib/src/defaults.rs:86]: If this is the default, then if I enable authentication, will I have to use this exact string? If it is arbitrary, could we make it simply `FuelLabs` instead?

**Review by lostman — DISMISSED** [review]

File: packages/fuel-indexer-lib/src/defaults.rs (hunk 1 above)
Comment (ra0x3) [reply packages/fuel-indexer-lib/src/defaults.rs:86]: @lostman 
- Short answer is we can make this string whatever we want
- Longer answer is this ☝🏽 particular string was my (possibly failed) attempt to show "hey, this needs to be a long secret string with a mix of characters"

Comment (lostman) [reply packages/fuel-indexer-lib/src/defaults.rs:86]: That's fine. I was thinking that if I was testing anything related to authentication locally, `FuelLabs` would've been nicer. That's the only reason I mentioned it.

**Review by lostman — DISMISSED** [review]

File: packages/fuel-indexer-macros/src/indexer.rs [hunk 2]
```diff
@@ -352,6 +352,13 @@ fn process_fn_items(
 
                 let fn_name = &fn_item.sig.ident;
 
+                if arg_list.is_empty() {
+                    proc_macro_error::abort_call_site!(
+                        "Handler function '{}' must have at least one argument.",
+                        fn_name.to_string(),
+                    );
+                }
```
Comment (deekerno) [inline packages/fuel-indexer-macros/src/indexer.rs:360]: 👌🏽

**Review by deekerno — APPROVED** [review]
Very nice work. 🥇 I especially like the insta tests for the help output.

**Pull request merged**